cpp/src/Ice/RequestHandlerFactory.cpp

```diff
@@ -52,7 +52,15 @@ IceInternal::RequestHandlerFactory::getRequestHandler(const RoutableReferencePtr
     }
     if(connect)
     {
-        ref->getConnection(handler);
+        ref->getConnectionAsync(
```

Comment:
I don't find this very elegant - the response/exception that update some other object (here, the ConnectRequestHandler).

I think the preferred pattern should be: if a call updates asynchronously an object state, it should be on that object. So here it would be instead:

```
handler->getConnectionAsync();
```

and handler in turn can call ref->getConnectionAsync(...) with response/exception that update its own state.